packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -37,28 +54,33 @@ export class Construct {
      * @param parent The parent construct
      * @param props  Properties for this construct
      */
-    constructor(parent: Construct, name: string) {
-        this.name = name;
+    constructor(parent: Construct, id: string) {
+        this.id = id;
+        this.name = id; // legacy
         this.parent = parent;
 
         // We say that parent is required, but some root constructs bypass the type checks and
         // actually pass in 'undefined'.
```

Comment:
And we allow that? 😱

More seriously, can't parent constructs pass a non-null parent in? Like... themselves?

Reply:
That's how it's implemented now.